backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -333,4 +334,42 @@ export class FeatureFlagService {
     const includedFeatureFlags = featureFlags.filter(({ id }) => includedFeatureFlagIds.includes(id));
     return includedFeatureFlags;
   }
+
+  public async validateImportFeatureFlags(featureFlagFiles: any, logger: UpgradeLogger): Promise<ValidatedFeatureFlagsError[]> {
+    logger.info({ message: 'Validate feature flags' });
+    const validationErrors = await Promise.allSettled(
+      featureFlagFiles.map(async (featureFlagFile) => {
+        let featureFlag = JSON.parse(featureFlagFile.fileContent);
+        // const newFeatureFlag = plainToClass(FeatureFlag, featureFlag);
+        const error = this.validateImportFeatureFlag(featureFlagFile.fileName, featureFlag);
+        return error;
+      })
+    );
+    // Filter out the files that have no promise rejection errors
+    return validationErrors
+      .map((result) => {
+        if (result.status === 'fulfilled') {
+          return result.value;
+        } else {
+          const { fileName, compatibilityType } = result.reason;
+          return { fileName: fileName, compatibilityType: compatibilityType };
+        }
+      })
+      .filter((error) => error !== null);
+  }
+
+  private validateImportFeatureFlag(fileName: string, flag: any) {
```

Comment:
We can type the `flag` parameter here if we're catching the parse failures in the `validateImportFeatureFlags()` method.

Reply:
code updated